src/util/semaphore.js

```diff
@@ -65,6 +65,17 @@ class Semaphore {
         }
     }
 
+    /**
+     * submit request without waiting
+     * @param {*} count
+     * @param {*} func
+     */
+    submit(count, func) {
```

Comment:
Didn't we want to drop calling func if the sem is full?

Reply:
@guymguym What do you mean "the submitted function will take care of releasing the acquired count."? this.release() inside the finally block will release the count. The submitted function does not need to do that.

Reply:
no, off course, you got it right,
I just used this to refer to the function we send to setImmediate(...)